packages/grafana-data/src/transformations/transformers/fillGaps.ts

```diff
@@ -0,0 +1,61 @@
+import { MutableDataFrame, DataFrame, FieldType } from '../..';
+
+import { map } from 'rxjs/operators';
+
+import { DataTransformerID } from './ids';
+import { DataTransformerInfo } from '../../types/transformations';
+
+export interface FillGapsTransformerOptions {
+  interval?: number;
+  fillWith: any;
+}
+
+export const fillGapsTransformer: DataTransformerInfo<FillGapsTransformerOptions> = {
+  id: DataTransformerID.fillGaps,
+  name: 'Fill Gaps',
+  description: '',
+  defaultOptions: {
+    fillWith: null,
+  },
+  operator: (options) => (source) =>
+    source.pipe(
+      map((dataFrames) => {
+        if (!Array.isArray(dataFrames)) {
+          return dataFrames;
+        }
+        return dataFrames.map((f) => fillGaps(f, options));
+      })
+    ),
+};
+
+const getInterval = (frame: DataFrame) => frame.fields.map((f) => f.config.interval ?? 0).find((i) => i > 0) ?? 0;
+
+export const fillGaps = (df: DataFrame, options?: FillGapsTransformerOptions): DataFrame => {
+  const frame = new MutableDataFrame(df);
+  const interval = options?.interval ?? getInterval(frame);
+  if (interval === 0) {
+    return frame;
+  }
+
+  const timeField = frame.fields.find((f) => f.type === FieldType.time);
```

Comment:
```suggestion
  const intervalField = frame.fields.find((f) => f.config.interval != null);
```

If we need to force it to be a time, can that be an argument?

Reply:
updated to use the first matching field with interval set